tofhir-server/src/main/scala/io/tofhir/server/service/job/JobFolderRepository.scala

```diff
@@ -136,9 +136,23 @@ override def getJob(projectId: String, id: String): Future[Option[FhirMappingJob
     })
   }
 
+  /**
+   * Retrieves the jobs referencing the given mapping in their definitions.
+   *
+   * @param projectId  identifier of project whose jobs will be checked
+   * @param mappingUrl the url of mapping
+   * @return the jobs referencing the given mapping in their definitions
+   */
+  override def getJobsReferencingMapping(projectId: String, mappingUrl: String): Future[Seq[FhirMappingJob]] = {
+    Future {
+      val jobs: Seq[FhirMappingJob] = jobDefinitions.getOrElse(key = projectId, default = Map.empty).values.toSeq
+      jobs.filter(job => job.mappings.map(mappingTask => mappingTask.mappingRef).contains(mappingUrl))
```

Comment:
Why do we use contains here? Should not we use equals or contentEquals?

Reply:
This method returns job list referencing specific mapping. That's why, we filter out those that do not contain the mapping url.